src/summary.rs

```diff
@@ -62,7 +64,24 @@ impl Summary {
         msg: &Message,
         chat: &Chat,
         contact: Option<&Contact>,
-    ) -> Self {
+    ) -> Result<Summary> {
+        if let Some((reaction_msg, reaction_contact_id, reaction)) = chat
+            .get_last_reaction_if_newer_than(context, msg.get_timestamp())
+            .await?
+        {
+            // there is a reaction newer than the latest message, show that.
+            // sorting and therefore date is still the one of the last message,
+            // the reaction is is more sth. that overlays temporarily.
+            let summary = reaction_msg.get_summary_text(context).await;
+            return Ok(Summary {
+                prefix: None,
+                text: msg_reacted(context, reaction_contact_id, &reaction, &summary).await,
+                timestamp: msg.get_timestamp(), // message timestamp (not reaction) - otherwise sorting is wrong
```

Comment:
Btw, sorting can be wrong anyway because we order chats in the chatlist by the sort timestamp, but display the "sent" timestamp. So, maybe we don't need to protect from this because it happens sometimes anyway, and just display the reaction time

Reply:
true. however, this "wrong message timestamp", messing up descending timestamp column, is the _exception in case of errors_.

if we would do that for  magnitudes more frequent reactions,  the timestamp column will appear unpredictable random instead of descending to the user. avoiding to get oriented there wrt newer/older up/down.

i think, this is also the main reason reason not to resort the list on reactions - it is just too much noise.